gcsfs/core.py

```diff
@@ -115,9 +115,13 @@ def _location():
     valid http location
     """
     _emulator_location = os.getenv("STORAGE_EMULATOR_HOST", None)
-    return (
-        _emulator_location if _emulator_location else "https://storage.googleapis.com"
-    )
+    if _emulator_location:
+        if not any(
+            _emulator_location.startswith(scheme) for scheme in ("http://", "https://")
+        ):
+            _emulator_location = f"http://{_emulator_location}"
```

Comment:
Good question - seems like `http` is assumed with third-party libraries only, since gcloud itself mostly uses grpc (I'm not too familiar with it myself). Here's an example from `gcloud.aio.pubsub`: https://github.com/talkiq/gcloud-aio/blob/master/pubsub/gcloud/aio/pubsub/publisher_client.py#L36

As for the format itself - the same convention seems to be followed across different projects, e.g.:
- Firestore: `FIRESTORE_EMULATOR_HOST` - https://firebase.google.com/docs/emulator-suite/connect_firestore#admin_sdks
  - Examples: https://github.com/googleapis/python-firestore/blob/a2bdb14768cfb366ca6ff8d80c66f3509134b5f0/owlbot.py, https://github.com/googleapis/python-firestore/blob/a8adf1eeaa361574c0d86918c8c87962034b42e6/noxfile.py#L272
- Pubsub: https://cloud.google.com/pubsub/docs/emulator#manually_setting_the_variables
- gcloud.aio.storage (third-party library): https://github.com/talkiq/gcloud-aio/blob/master/storage/gcloud/aio/storage/storage.py#L47

Reply:
...on the other hand, official library also seems to violate this convention:

https://github.com/googleapis/python-storage/blob/main/google/cloud/storage/_helpers.py#L43

Reply:
😃